sdk/core/azure-core/inc/azure/core/context.hpp

```diff
@@ -245,10 +368,14 @@ namespace Azure { namespace Core {
       }
     }
 
-    /**
-     * @brief The application context (root).
+    /** @brief The `ApplicationContext` is the root of all Context objects.
+     *
+     * @note: The `ApplicationContext` object is deprecated and will be removed in a future release.
+     * If your application is using `ApplicationContext`, you should create your own root context
+     * and us it where you would have otherwise used `ApplicationContext`.
      *
      */
-    static const AZ_CORE_DLLEXPORT Context ApplicationContext;
+    [[deprecated("ApplicationContext is no longer supported. Instead customers should create their "
+                 "own root context objects.")]] static AZ_CORE_DLLEXPORT Context ApplicationContext;
```

Comment:
Leave it as const, rather than introducing another break since we are already deprecating it?
```suggestion
                 "own root context objects.")]] static const AZ_CORE_DLLEXPORT Context ApplicationContext;
```